ViewLayer/SongPlayerView/SongPlayerView.cpp

```diff
@@ -45,9 +46,12 @@ void SongPlayerView::handleVolumeControl()
     songPlayer_.adjustVolume(volume);
 }
 
-void SongPlayerView::updateGUI(const QString& title, const QString& artist)
+void SongPlayerView::updateGUI(const QString& title, const QString& artist, const QString& cover)
 {
     ui_.infoLabel().setText(artist + " - " + title);
+    QPixmap img(cover);
```

Comment:
Let's make `cover` be an actual `QPixmap` over at `SongPlayer`, then here we should be checking `cover` to see if it's null or not, if so, display the default background.